pkg/grpc/grpc.go

```diff
@@ -92,6 +93,9 @@ func NewGRPCServersFromConfigurationAndServe(configurations []*configuration.GRP
 		grpc_prometheus.EnableHandlingTimeHistogram(grpc_prometheus.WithHistogramBuckets(prometheus.DefBuckets))
 		grpc_prometheus.Register(s)
 
+		// Enable grpc reflection.
+		reflection.Register(s)
```

Comment:
Does reflection honour authentication? As in, what happens if you set the authentication policy to ‘deny’? Does reflection then still work?

If that’s the case, there might be some security implications to it. You’d be able to sniff that an arbitrary TCP port on the internet and discover it’s Buildbarn. Then we’d better put it behind a Boolean configuration flag.

Reply:
Looks like it does honor it. After setting 
```
  "grpcServers": [{
    "listenAddresses": [":8981"],
    "authenticationPolicy": { "deny": "go away" }
  }],
```

it behaves like so:
```
$ grpcurl -plaintext localhost:8981 list
Failed to list services: rpc error: code = Unauthenticated desc = go away

$ grpcurl -plaintext localhost:8981 describe build.bazel.remote.execution.v2.ActionCache
Failed to resolve symbol "build.bazel.remote.execution.v2.ActionCache": rpc error: code = Unauthenticated desc = go away
```

Reply:
🎉